spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -622,8 +590,7 @@ private static class UnpartitionedDeltaWriter extends DeleteAndDataDeltaWriter {
 
     @Override
     public void update(InternalRow meta, InternalRow id, InternalRow row) throws IOException {
-      delete(meta, id);
-      delegate.update(row, dataSpec, null);
+      throw new UnsupportedOperationException("Update must be represented as delete and insert");
```

Comment:
This makes me happy that we actually write updates as separate deletes and inserts.